### PR TITLE
converts cows to basic mobs and ADDS MOONICORNS

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Cows are now basic mobs, pretty elementary stuff. Their speed may be slightly different because I couldn't convert `turns_per_move` into `speed` (and trust me i tried)

also, MOONICORNS. They attack with their beautiful unicorn horn, which pacifies on hit. As they walk they spread fairygrass. Aren't they wonderful? As it turns out, no, they're actually incredibly aggressive and their pacification is incredibly deadly as a result.

If you hold something they love, they'll stop attacking. Use this power for good and not evil!

## Why It's Good For The Game
<details>
  <summary>Spoiler warning</summary>
  
  image
  
</details>

okay but in all seriousness Moonicorns were a great encouragement to convert another mob to basic mobs while adding some dangerous cyto rewards.

## Changelog

:cl:
add: MOONICORNS. Watch out for their horn!
refactor: during a code refactor for cow, they may be SLIGHTLY faster or SLIGHTLY slower as an accidental result
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Is charging a datum AI behavior yet? If it is I would recommend giving it to the moonicorns. They've got a giant spike rooted to their forehead.

**Comment by TemporalOroboros** [comment]
Also, the left-facing moonicorn sprite appears to be missing the horn.

**Comment by AMonkeyThatCodes** [comment]
can we please not mix refactors and content additions for basic mobs?

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -118,7 +118,7 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
 #define ismouse(A) (istype(A, /mob/living/simple_animal/mouse))
 
-#define iscow(A) (istype(A, /mob/living/simple_animal/cow))
+#define iscow(A) (istype(A, /mob/living/basic/cow))
```
Comment (AMonkeyThatCodes) [inline code/__DEFINES/is_helpers.dm:121]: why does this exist

Comment (AMonkeyThatCodes) [reply code/__DEFINES/is_helpers.dm:121]: ok i checked its for one use by abductors. it made me laugh but its still dumb

File: code/modules/mob/living/basic/farm_animals/cows.dm [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+/mob/living/basic/cow/proc/after_cow_tipped(mob/living/carbon/tipper)
+	addtimer(CALLBACK(src, .proc/look_for_help, tipper), rand(10 SECONDS, 20 SECONDS))
+
+/*
+ * Find a mob in a short radius around us (prioritizing the person who originally tipped us)
+ * and either look at them for help, or give up. No actual mechanical difference between the two.
+ *
+ * tipper - the mob who originally tipped us
+ */
+/mob/living/basic/cow/proc/look_for_help(mob/living/carbon/tipper)
```
Comment (AMonkeyThatCodes) [inline code/modules/mob/living/basic/farm_animals/cows.dm:71]: this is AI code, dont put it in the mob.

**Comment by tralezab** [comment]
moonicorns are necessary for this pr because basic mobs are not fun to make

**Comment by AMonkeyThatCodes** [comment]
just split things up in two prs in the future, its like a miniature extra step.

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		var/text = pick("imploringly", "pleadingly", "with a resigned expression")
+		seen_message = "[controller.pawn] looks at [savior] [text]."
+		self_message = "You look at [savior] [text]."
+	else
+		seen_message = "[controller.pawn] seems resigned to its fate."
+		self_message = "You resign yourself to your fate."
+	controller.pawn.visible_message(span_notice("[seen_message]"), span_notice("[self_message]"))
+	finish_action(controller, TRUE)
+
+/datum/ai_behavior/tipped_reaction/finish_action(datum/ai_controller/controller, succeeded)
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:33]: Finish action should have the Same optional arguments as perform (tipper_key at the end)

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	. = ..()
+	//I'VE SAID MY PEACE...
+	controller.blackboard[BB_BASIC_MOB_TIP_REACTING] = FALSE
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:36]: Reset the tipper too

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Fikou** [comment]
moo

**Comment by tralezab** [comment]
moo

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm [hunk 5]
```diff
@@ -28,9 +28,10 @@
...
 	controller.pawn.visible_message(span_notice("[seen_message]"), span_notice("[self_message]"))
-	finish_action(controller, TRUE)
+	finish_action(controller, TRUE, tipper_key)
 
-/datum/ai_behavior/tipped_reaction/finish_action(datum/ai_controller/controller, succeeded)
+/datum/ai_behavior/tipped_reaction/finish_action(datum/ai_controller/controller, succeeded, tipper_key)
 	. = ..()
 	//I'VE SAID MY PEACE...
 	controller.blackboard[BB_BASIC_MOB_TIP_REACTING] = FALSE
+	controller.blackboard[BB_BASIC_MOB_TIPPER] = null
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:37]: Use the argument ya filthy dog

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm [hunk 6]
```diff
@@ -28,9 +28,10 @@
...
 		self_message = "You resign yourself to your fate."
 	controller.pawn.visible_message(span_notice("[seen_message]"), span_notice("[self_message]"))
-	finish_action(controller, TRUE)
+	finish_action(controller, TRUE, tipper_key)
 
-/datum/ai_behavior/tipped_reaction/finish_action(datum/ai_controller/controller, succeeded)
+/datum/ai_behavior/tipped_reaction/finish_action(datum/ai_controller/controller, succeeded, tipper_key)
 	. = ..()
 	//I'VE SAID MY PEACE...
 	controller.blackboard[BB_BASIC_MOB_TIP_REACTING] = FALSE
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:36]: Make this an argument ya filthy dog

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm (hunk 5 above)
Comment (tralezab) [reply code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:37]: I DONT SEE MUCH POINT but ok

File: code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm (hunk 6 above)
Comment (tralezab) [reply code/datums/ai/basic_mobs/basic_ai_behaviors/tipped_reaction.dm:36]: I DONT SEE MUCH POINT but ok

**Pull request merged**